fix: check for y vector order in `bin`

RogerGinBer opened the pull request:
Add a check to `bin` to ensure y is ordered. #108
Update related documentation.
Add unit tests to ensure the fix works.

sgibb commented on R/binning.R:
```diff
@@ -9,7 +9,8 @@
 #'
 #' @param y `numeric` with same length than `x` with values to be used for
-#'     the binning.
+#'     the binning. Ideally, `y` should be increasingly sorted, but it's not 
+#'     mandatory.
```
I am somehow against silently reordering `y`. Please see https://github.com/rformassspectrometry/MsCoreUtils/issues/108#issuecomment-1382383808

lgatto replied:
I'm with @sgibb here and advise against silently re-ordering.

jorainer replied:
Yes, agree. Please no silent re-ordering (see also https://github.com/rformassspectrometry/MsCoreUtils/issues/108#issuecomment-1400453934)

sgibb commented on tests/testthat/test_binning.R:
```diff
@@ -48,6 +48,16 @@ test_that("bin works", {
 
+    ## Ensure y order does not matter
+    ## Issue #108 https://github.com/rformassspectrometry/MsCoreUtils/issues/108
+    expect_false(is.unsorted(xs))
```
No need to check that base R functions are working. 
```suggestion
```

sgibb commented on tests/testthat/test_binning.R:
```diff
@@ -48,6 +48,16 @@ test_that("bin works", {
+    expect_false(is.unsorted(xs))
+    expect_equal(bin(vals, xs), bin(rev(vals), rev(xs)))
+    xs_unsorted <- sample(xs, length(xs))
+    expect_true(is.unsorted(xs_unsorted))
```
```suggestion
```

sgibb commented on tests/testthat/test_binning.R:
```diff
@@ -48,6 +48,16 @@ test_that("bin works", {
     expect_true(is.numeric(res_2))
     expect_equal(res$x, res_2)
 
+    ## Ensure y order does not matter
```
I know it is done above as well but why we need such a "complex" unit test if I could simply test `bin(1:3, 1:3)` vs `bin(3:1, 3:1)`.

Unit tests should be as easy as possible and should require nearly no runtime (I know we are not good at both in all RforMassSpectrometry packages).

sgibb reviewed (changes requested):
Thanks for the effort of creating a PR. While @jorainer asked for it I like to discuss first if we really want to reorder `y` in #108. If we really agree that it should be done, we have to ensure that `breaks` are ordered as well.

jorainer commented:
Agree with @sgibb . I would keep this function simple and throw an error if `x` is not ordered. This test could be made conditional with the `.check` parameter (default `.check = TRUE`) which would allow skipping it for data where it is known that the values are sorted.

@RogerGinBer , can you please also describe this requirement in the documentation? This function is supposed to allow binning of m/z values of a single spectrum and these should be increasingly ordered by definition.

RogerGinBer commented:
Hi there, I've removed the silent reordering, created a `.check` param (default `TRUE`) to throw an error if `y` is unordered, and updated tests and docs accordingly. I think that should be all, let me know if we need anything else :+1:

jorainer reviewed (changes requested):
Thanks Roger @RogerGinBer for the PR! Only thing I would ask you to do it to add you as a contributor (`"ctb"`) to the `DESCRIPTION`.

Thanks!

jorainer commented:
Thanks @RogerGinBer ! I'll merge now and push to BioC
